client/common/configuration/templates/ChannelConfigurationElement.ts

```diff
@@ -0,0 +1,332 @@
+import { RegisterClientOptions } from '@peertube/peertube-types/client'
+import { css, html, LitElement } from 'lit'
+import { repeat } from 'lit-html/directives/repeat.js'
+import { customElement, property, state } from 'lit/decorators.js'
+import { ptTr } from './TranslationDirective'
+import { localizedHelpUrl } from '../../../utils/help'
+import './DynamicTableFormElement'
+import './PluginConfigurationRow'
+import './HelpButtonElement'
+import { until } from 'async'
+import { Task } from '@lit/task';
+import { ChannelConfiguration } from 'shared/lib/types'
+import { ChannelConfigurationService } from './ChannelConfigurationService'
+import { createContext, provide } from '@lit/context'
+import { getGlobalStyleSheets } from '../../global-styles'
+
+export const registerClientOptionsContext = createContext<RegisterClientOptions | undefined>(Symbol('register-client-options'));
+export const channelConfigurationContext = createContext<ChannelConfiguration | undefined>(Symbol('channel-configuration'));
+export const channelConfigurationServiceContext = createContext<ChannelConfigurationService | undefined>(Symbol('channel-configuration-service'));
+
+@customElement('channel-configuration')
+export class ChannelConfigurationElement extends LitElement {
+
+  @provide({ context: registerClientOptionsContext })
+  @property({ attribute: false })
+  public registerClientOptions: RegisterClientOptions | undefined
+
+  @property({ attribute: false })
+  public channelId: number | undefined
+
+  @provide({ context: channelConfigurationContext })
+  @state()
+  public _channelConfiguration: ChannelConfiguration | undefined
+
+  @provide({ context: channelConfigurationServiceContext })
+  private _configurationService: ChannelConfigurationService | undefined
+
+  static styles = [
+    ...getGlobalStyleSheets()
```

Comment:
Could such code be moved in a parent class? So we don't have to add it in each element class.

Something like `LivechatLitElement`?